src/texteff.cpp

```diff
@@ -36,39 +36,40 @@ struct TextEffect : public ViewportSign {
 	}
 };
 
-static SmallVector<struct TextEffect, 32> _text_effects; ///< Text effects are stored there
+static std::vector<struct TextEffect> _text_effects; ///< Text effects are stored there
 
 /* Text Effects */
 TextEffectID AddTextEffect(StringID msg, int center, int y, uint8 duration, TextEffectMode mode)
 {
 	if (_game_mode == GM_MENU) return INVALID_TE_ID;
 
-	TextEffectID i;
-	for (i = 0; i < _text_effects.Length(); i++) {
-		if (_text_effects[i].string_id == INVALID_STRING_ID) break;
+	auto it = _text_effects.begin();
+	for (; it < _text_effects.end(); ++it) {
```

Comment:
could probably be merged onto a single line

Reply:
`it` needed to persist beyond the scope of the for loop. However, this is basically `std::find_if`, so I would re-write it with that instead.